src/lib.rs

```diff
@@ -779,14 +780,15 @@ mod tests {
             assert_eq!(w.fsck().unwrap(), 3);
         }
 
-        let read_manifest = w.read_manifest().unwrap();
+        let idx = w.read_index()?.unwrap();
```

Comment:
Could you remove these `unwrap`s  and return errors instead?

Reply:
This is in test code where I often use unwrap so we get the exact line for errors.

Reply:
Though "why use both ? and unwrap()"? Basically because `.unwrap().unwrap()` is longer and visually ugly, and I don't actually expect I/O errors in the tests, but it's quite possible that we somehow fail to return a `Some()`.

Reply:
Sorry, didn't notice it was in tests. One thing that I like is to return `anyhow::Result` from my tests, this way I can use `?` consistently instead of having to use `unwrap`.

Reply:
In this case we have `Result<Option<>>` though so we can't `??` today anyways.

Reply:
Yeah, we would need `?.ok_or(...)?` / `?.ok_or_else(...)?`.